### PR TITLE
Reward classifier and training

### PR DESCRIPTION
**Pull request by ChorntonYoel**
## What this does
|  Title               | Label           |
|----------------------|-----------------|
| Item 2 of Reward Classifier in issue #504       |  (Feature)        |

This PR is meant to add a reward classifier (used to classify if an image of a robot performing a task should get a reward or not), a training file allowing the training of the classifier (with logging + resuming), a config.yaml file that can be used  to start a training, and a few tests for the training loop

## How it was tested
Using 10 episodes made with the reward system of this PR: https://github.com/huggingface/lerobot/pull/518
Also I added a test file for the training classifier file. Lots of things are mocked but it covers the basics I believe.

## How to checkout & try? (for the reviewer)
```
python lerobot/scripts/train_classifier.py \
    --config-name", "policy/reward_classifier.yaml",
```
With  the wandb entity and the dataset name adapted.

I was able to reproduce 95%+ after a few epochs with `facebook/convnext-base-224` as backbone and a dataset of 10 epsiodes of ~15 sec.
This branch was built on top of the branch from #518 so will need to wait for this one to be merged befre merging

### CODE REVIEW

**Comment by michel-aractingi** [comment]
Great work! looking forward to trying it, let me know when I can review it.

File: tests/test_train_classifier.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
+# test_train_classifier.py
```
Comment (Cadene) [inline tests/test_train_classifier.py:1]: Nice file!

Out of curiosity, why this comment `# test_train_classifier.py`?

File: lerobot/configs/policy/reward_classifier.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# @package _global_
```
Comment (Cadene) [inline lerobot/configs/policy/reward_classifier.yaml:1]: Nice file!

File: lerobot/scripts/control_robot.py [hunk 3]
```diff
@@ -218,6 +219,7 @@ def record(
     policy = None
     device = None
     use_amp = None
+    features = {"next.reward": {"dtype": "int64", "shape": (1,), "names": None}} if assign_rewards else None
```
Comment (Cadene) [inline lerobot/scripts/control_robot.py:222]: Maybe?
```suggestion
    extra_features = {"next.reward": {"dtype": "int64", "shape": (1,), "names": None}} if assign_rewards else None
```

File: tests/test_train_classifier.py (hunk 1 above)
Comment (ChorntonYoel) [reply tests/test_train_classifier.py:1]: No good reason haha force of habit, for some pre-commit setups you need a comment at the beginning of the file, so I start by a dummy one. will take off

**Comment by michel-aractingi** [comment]
Nice work @ChorntonYoel ! Could you move the classifier directory to `lerobot/common/policies/classifier/` to `lerobot/common/policies/hilserl/classifier`.

Since now we will only use the reward classifier for `hil-serl` then we will put everything in its directory. In the future when the classifiers are more established we can have a separate directory in `lerobot/common/classifiers` to host different kinds of reward recognition models.

File: lerobot/common/robot_devices/control_utils.py [hunk 4]
```diff
@@ -120,14 +120,16 @@ def predict_action(observation, policy, device, use_amp):
     return action
 
 
-def init_keyboard_listener():
+def init_keyboard_listener(assign_rewards=False):
     # Allow to exit early while recording an episode or resetting the environment,
     # by tapping the right arrow key '->'. This might require a sudo permission
     # to allow your terminal to monitor keyboard events.
```
Comment (michel-aractingi) [inline lerobot/common/robot_devices/control_utils.py:126]: I made this comment on #518 but I'll add it it here since this PR contains all the additions of the previous one. 
Should we add some comments here to explain `assign_rewards`?
```suggestion
    """
    Initializes a keyboard listener to enable early termination of an episode 
    or environment reset by pressing the right arrow key ('->'). This may require 
    sudo permissions to allow the terminal to monitor keyboard events.

    Args:
        assign_rewards (bool): If True, allows annotating the collected trajectory 
        with a binary reward at the end of the episode to indicate success.
    """
```

File: lerobot/common/datasets/lerobot_dataset.py [hunk 5]
```diff
@@ -291,7 +291,8 @@ def create(
         obj.root.mkdir(parents=True, exist_ok=False)
 
         if robot is not None:
-            features = get_features_from_robot(robot, use_videos)
+            features = {} if features is None else features
+            features.update(get_features_from_robot(robot, use_videos))
```
Comment (michel-aractingi) [inline lerobot/common/datasets/lerobot_dataset.py:295]: This comment was also made on #518 . 
What do you think of writing it like this?
```suggestion
            features = {**(features or {}), **get_features_from_robot(robot)}
```

File: lerobot/common/robot_devices/control_utils.py (hunk 4 above)
Comment (ChorntonYoel) [reply lerobot/common/robot_devices/control_utils.py:126]: ah yes that's nice

File: lerobot/common/policies/factory.py [hunk 6]
```diff
@@ -66,6 +66,11 @@ def get_policy_and_config_classes(name: str) -> tuple[Policy, object]:
         from lerobot.common.policies.vqbet.modeling_vqbet import VQBeTPolicy
 
         return VQBeTPolicy, VQBeTConfig
+    elif name == "classifier":
+        from lerobot.common.policies.classifier.configuration_classifier import ClassifierConfig
+        from lerobot.common.policies.classifier.modeling_classifier import Classifier
+
+        return Classifier, ClassifierConfig
```
[the remark this answers is not in the document]
Comment (ChorntonYoel) [reply lerobot/common/policies/factory.py]: Is this okay now that the classifier has the "policy" hilserl/classifier" ? 
Or do you still think it's confusing and we should initialize it in a different way?

File: lerobot/scripts/train_classifier.py [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+        cfg = checkpoint_cfg
+        cfg.resume = True
+
+    # Initialize model and training components
+    model = make_policy(
+        hydra_cfg=cfg,
+        dataset_stats=dataset.meta.stats if not cfg.resume else None,
+        pretrained_policy_name_or_path=str(logger.last_pretrained_model_dir) if cfg.resume else None,
+    ).to(device)
+
```
[the remark this answers is not in the document]
Comment (ChorntonYoel) [reply lerobot/scripts/train_classifier.py]: Outdated, but would you still prefer I do that?

File: lerobot/common/policies/factory.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ChorntonYoel) [reply lerobot/common/policies/factory.py]: done
